### PR TITLE
better android push handling

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this fixes push handling when you've backed out into the background

### CODE REVIEW

File: shared/actions/platform-specific.native.js [hunk 1]
```diff
@@ -97,8 +115,31 @@ function displayNewMessageNotification(text: string, convID: ?string, badgeCount
   })
 }
 
+let lastPush = null
 function configurePush() {
   return eventChannel(dispatch => {
+    if (RNEmitter) {
+      // If android launched due to push
+      RNEmitter.addListener('androidIntentNotification', payload => {
```
Comment (akalin-keybase) [inline shared/actions/platform-specific.native.js]: payload parameter not used?

File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java [hunk 2]
```diff
@@ -69,6 +71,13 @@ protected void onCreate(Bundle savedInstanceState) {
 
         super.onCreate(savedInstanceState);
 
+        ReactContext currentContext = getReactInstanceManager().getCurrentReactContext();
+        if (currentContext != null) {
+            currentContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                    .emit("androidIntentNotification", getIntent().getExtras().containsKey("notification"));
```
Comment (akalin-keybase) [inline shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java]: we pass a boolean parameter here, but we don't seem to use it when consuming `androidIntentNotification`. Dead code?

File: shared/actions/platform-specific.native.js (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/platform-specific.native.js]: oh yeah, lemme move that check into the java side

File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java [hunk 3]
```diff
@@ -69,6 +71,15 @@ protected void onCreate(Bundle savedInstanceState) {
 
         super.onCreate(savedInstanceState);
 
+        if (getIntent().getExtras().containsKey("notification")) {
+            ReactContext currentContext = getReactInstanceManager().getCurrentReactContext();
+            if (currentContext != null) {
+                currentContext
+                        .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                        .emit("androidIntentNotification", "");
```
Comment (akalin-keybase) [inline shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java]: `null` better than `""`? https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/core/DeviceEventManagerModule.java seems to indicate `null` is ok

**Pull request merged**